controllers/front/listing/CategoryController.php

```diff
@@ -87,10 +88,40 @@ public function init()
         parent::init();
 
         if (!Validate::isLoadedObject($this->category) || !$this->category->active || !$this->category->existsInShop($this->context->shop->id)) {
-            header('HTTP/1.1 404 Not Found');
-            header('Status: 404 Not Found');
-            $this->setTemplate('errors/404');
-            $this->notFound = true;
+            if (!$this->category->id_type_redirected) {
+                if (in_array($this->category->redirect_type, [RedirectType::TYPE_CATEGORY_PERMANENT, RedirectType::TYPE_CATEGORY_TEMPORARY])) {
+                    $this->category->id_type_redirected = $this->category->id_category_default;
```

Comment:
Careful I wouldn't trust this field, it is never written/read It has not link to the DB and no domain logic behind (no default category is assigned to a category) I don't know its purpose but I'm not sure it is what you are looking for

In the form we stated that empty value is equivalent to the root category, so I think you need to fetch the root category another way, I don't know if it's in a configuration or if you need to do an SQL request to fetch it though

Reply:
Maybe you can use `Category::getRootCategory()->id`

Reply:
An alternative I just though about would be to redirect to the parent when the value is left empty, but that's a product decision

Reply:
@MatShir ?

Reply:
The parent category could be disabled as well. Is it going to redirect to the parent until a parent category is available ?

Reply:
I used the getRootCategory, which works.

For `$this->category->id_type_redirected = $this->category->id_category_default;`, this is code taken from the Product controller, so the remark also applies to the product side